crates/evm/core/src/backend/strategy.rs

```diff
@@ -181,6 +196,48 @@ impl BackendStrategyRunner for EvmBackendStrategyRunner {
     ) {
         EvmBackendMergeStrategy::merge_db_account_data(addr, active, fork_db);
     }
+
+    // Note(zk): This function in upstream code is not implemented as part of the strategy pattern,
+    // but is instead a standalone function. We have moved it here to make it part of the
+    // strategy, as there is some abstraction in the middle since the envelopes, and types are
+    // different. The changes are:
+    // - The function signature has been changed to take a `Bytes` instead of `the
+    //   TransactionRequest`
+    // - The function signature has been changed to return a `TransactionMaybeSigned` instead of
+    //   empty tuple
+    // - Avoids some cloning of the backend state
+    // See the main function in entrypoint: crates/evm/core/src/backend/mod.rs
+    // See zk implementation in: crates/strategy/zksync/src/backend/runner.rs
```

Comment:
Being part of the strategy, it's already implicit upstream doesn't implement it
```suggestion
    // NOTE(zk): This is the equivalent of the original EVM implementation for `transact_from_tx`, with some backend cloning optimization.
```